XLDO-37 & XLDO-38 Improvements for XL JetPack Docker Image

ielshal opened the pull request:
- Added new flag "--pull-registry" for rendering to specify Registry to pull XebiaLabs base Docker image from by default it will use xebialabs.
- This is needed for generating XLJetPack images especially during nightly as alpha versions are not pushed to official docker hub registry xebialabs hence we need a way to specify different registry like xebialabsunsupported etc.
- Added new context for XLJetpack conf to use "base_product" which is passed to docker file template.
- Added support for generating Image for CentOS.
- Added more details in Readme file including new flag added and xl jetpack images.
- Tested those changes locally without issues.

ielshal commented on README.adoc:
```diff
@@ -47,7 +47,14 @@ If you want to generate the Dockerfiles for a specific components of the XebiaLa
+
+[source,shell,subs="verbatim,attributes"]
+----
+$ {executable} render --xl-version <version> --product xl-jetpack-deploy --pull-registry xebialabsunsupported
```
If we are sure that everytime it will be the same repo, i mean one you pull from and push too .. which might be the case now then yes .. but my idea is to make it more flexible for future use too.

For example .. if you want to pull from xebialabsunsupported, build image and push to local registry xl-docker.xebialabs.com or the otherway around.
Also for other users like if they wants to pull from xebialabs registry and push to their local one.

Also my suggestion is to have 2 flags here --pull-registry (which i have this pull request opened for) and --push-registry (which can be just rename of --registry)

hierynomus replied:
Currently the flexibility is there, as they're 2 commands, `render` and `build` ;)

ielshal replied:
Aha okay got your point so you mean to use --registry flag, with render it will mean registry you want to pull from and with build it will mean registry you want to push to .. right ?

hierynomus replied:
Yes

ielshal replied:
Okay will do this .. Thanks 👍 :-)

ielshal commented:
Using `--registry` instead of `--pull-registry` with render as discussed with @hierynomus .

ielshal commented:
Added `--registry` to shared options as per @hierynomus recommendation.

ielshal commented:
* Added new flag "--registry" for rendering to specify Registry to pull XebiaLabs base Docker image from by default it will use xebialabs .. This flag will be used during render to pull images from registry passed or build for push images to it.
* Added new context for XLJetpack conf to use "base_product" which is passed to docker file template.
* Added support for generating xl-jetpack Image for CentOS.
* Added more details in Readme file including new flag added and xl jetpack images.
